Replace foundation tooltips with custom implementation

Simrayz opened the pull request:
## Scope and purpose

Part of #2972. 

Adds an implementation for a custom NAV Tooltip. This PR replaces usages of `[data-tooltip]` throughout the codebase, with the exception of one case in the Status tool that should be done in a follow-up task, namely #3463 .

The new tooltip supports alignment from the start or end of the trigger element, and automatic change of alignment if the tooltip extends beyond the viewport.

<!-- remove things that do not apply -->
### This pull request
* Adds nav tooltip styles and event listeners for updating alignment based on viewport
* Replaces usages of [data-tooltip] in various tools with the new implementation
* Replaces usage of [data-tooltip] in reusable form templates, like `foundation-5/field.html` or `custom_crispy_templates/field_helptext_as_icon.html`

## Screenshots

### Before
**/search/room/100/#!netboxinterfaces**
<img width="319" height="171" alt="image" src="https://github.com/user-attachments/assets/6e9ac0d0-3e3a-457a-9b29-ad77b93adb12" />

**/alertprofiles/profile/1026/**
<img width="459" height="182" alt="image" src="https://github.com/user-attachments/assets/5fdcfb80-a3f4-476c-a069-5595b5a79ee6" />

**/l2trace/**
<img width="518" height="267" alt="image" src="https://github.com/user-attachments/assets/78a852ae-a6a6-4cc7-826f-8096f273e37a" />

### After
**/search/room/100/#!netboxinterfaces**
<img width="313" height="155" alt="image" src="https://github.com/user-attachments/assets/dcb56249-d65d-44c6-90f2-60d555b45154" />

**/alertprofiles/profile/1026/**
<img width="406" height="182" alt="image" src="https://github.com/user-attachments/assets/51a9be3f-13dd-4287-86bd-d9b09b47aee7" />

**/l2trace/**
<img width="467" height="266" alt="image" src="https://github.com/user-attachments/assets/23aaa9b0-345b-4956-9282-8789c74e205c" />

## Contributor Checklist

Every pull request should have this checklist filled out, no matter how small it is.
More information about contributing to NAV can be found in the
[Hacker's guide to NAV](https://nav.readthedocs.io/en/latest/hacking/hacking.html#hacker-s-guide-to-nav).

<!-- Add an "X" inside the brackets to confirm -->
<!-- If not checking one or more of the boxes, please explain why below each or remove the line if not applicable. -->

* [x] Added a changelog fragment for [towncrier](https://nav.readthedocs.io/en/latest/hacking/hacking.html#adding-a-changelog-entry)
* [ ] Added/amended tests for new/changed code
* [ ] Added/changed documentation
* [x] Linted/formatted the code with ruff, easiest by using [pre-commit](https://nav.readthedocs.io/en/latest/hacking/hacking.html#pre-commit-hooks-and-ruff)
* [x] The first line of the commit message continues the sentence "If applied, this commit will ...", starts with a capital letter, does not end with punctuation and is 50 characters or less long. See https://cbea.ms/git-commit/
* [x] This pull request is based on the correct upstream branch: For a patch/bugfix affecting the latest stable version, it should be based on that version's branch (`<major>.<minor>.x`). For a new feature or other additions, it should be based on `master`.
* [ ] If applicable: Created new issues if this PR does not fix the issue completely/there is further work to be done
* [x] If this results in changes in the UI: Added screenshots of the before and after
* [ ] If this adds a new Python source code file: Added the [boilerplate header](https://nav.readthedocs.io/en/latest/hacking/hacking.html#python-boilerplate-headers) to that file

<!-- Make this a draft PR if the content is subject to change, cannot be merged or if it is for initial feedback -->

hmpf reviewed (changes requested):
Whitespace around the text is inconsistent.

Simrayz commented:
> Whitespace around the text is inconsistent.

Are you referring to the "l2trace" tooltip? Other than the width, this is identical to the foundation tooltip. The ul/li elements are affected by some global css, which I've not adjuster in this PR as I'm not sure of the side effects.

Simrayz commented:
> Whitespace around the text is inconsistent.

@hmpf I adjusted the whitespace (removing trailing margin) and font-size to make it consistent. Also added the bullet points to make it clearer that it is a list.
<img width="340" height="231" alt="image" src="https://github.com/user-attachments/assets/f0ca7190-a548-4604-9008-e39a8cd907f4" />
